.github/workflows/ci.yaml

```diff
@@ -32,27 +32,20 @@ jobs:
         uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
-      - name: Install Poetry
-        uses: snok/install-poetry@v1
-      - name: Load cached venv
-        id: cached-poetry-dependencies
-        uses: actions/cache@v3
+      - uses: pdm-project/setup-pdm@v3
+        name: Set up PDM
         with:
-          path: .venv
-          key: venv-${{ runner.os }}-${{ matrix.python-version }}-${{ hashFiles('**/poetry.lock') }}
+          python-version: ${{ matrix.python-version }}
```

Comment:
I think it's probably better to cache these as seen [here](https://github.com/litestar-org/polyfactory/blob/79360f18340da6ab67808a4a177349b206432bc9/.github/workflows/ci.yaml#L59-L64).